src/platforms/esp32/components/avm_builtins/i2c_driver.c

```diff
@@ -613,6 +628,59 @@ static NativeHandlerResult i2cdriver_consume_mailbox(Context *ctx)
     return cmd == I2CCloseCmd ? NativeTerminate : NativeContinue;
```

Comment:
I believe this now needs to be changed to prevent the native process to be deallocated while a "native i2c driver" locked the port.

Reply:
fixed